### ISSUE TITLE
Preview using nVim instead of `cat` or `bat`

### ISSUE THREAD

**Issue by savchenko** [linrongbin16/fzfx.nvim#491]
Is it technically possible? Having consistent colourscheme would be great.

**Comment by linrongbin16** [comment]
> Is it technically possible? Having consistent colourscheme would be great.

hi, it should be possible, but it requires a lot of effort, and I had once did a little investigation on the solution.

but now I still don't know how to do it.

the [fzf-lua](https://github.com/ibhagwan/fzf-lua) can do it, but I don't understand the source code.

**Comment by linrongbin16** [comment]
it's also one of the very important feature I want to add to it, for very long time.

I will look into it.

**Comment by linrongbin16** [comment]
the first buffer previewer work is at: https://github.com/linrongbin16/fzfx.nvim/pull/556

for now it can only preview file content (but I think it's the only major use case), without line number support (I will support line numbers for live grep in next PR).

**Comment by linrongbin16** [comment]
hi @savchenko , after #557 and #564 been merged into main branch, can you take a look if current bat previewer is good enough?

I will say it's still about 85% close to native nvim buffer. But the performance is without any defects.

**Comment by linrongbin16** [comment]
hi @savchenko , this issue could be the same issue with #575 , and I just fixed it in #576 .

Would you please pull main branch and try again? see if this issue still exists.

**Comment by linrongbin16** [comment]
> Updated again, the error is gone. Further setting `$BAT_THEME` to one generated by `fzfx.nvim` results in a reasonably close match:
> 
> ![02-Feb_20-07-35](https://private-user-images.githubusercontent.com/300146/301798446-0fbdf5c8-d1d8-4dee-b828-07e649203ccc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4Njc5MzQsIm5iZiI6MTcwNjg2NzYzNCwicGF0aCI6Ii8zMDAxNDYvMzAxNzk4NDQ2LTBmYmRmNWM4LWQxZDgtNGRlZS1iODI4LTA3ZTY0OTIwM2NjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwMlQwOTUzNTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNGQ4NDYxMmY4ZTg3N2Q2YzBhMzFhMDgwODM2YmI2ZDUyMDQ2Y2IzM2M3NTJjZjMwMTJmMDgxZjA2ZDI5YzIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.v1rFKsk_DCPHlQ-6V9C5R_NyE4020Wv0q5wDQirooKo)

It's automatically generated, you don't need to specify it in `BAT_THEME`

**Comment by linrongbin16** [comment]
I am trying to improve the converter (from nvim syntax to TextMate theme) again and again, to make it closer.

**Comment by linrongbin16** [comment]
> The issue with `(` breaking the search functionality is still present though.
> 
> Attaching `fzfx.log`: [fzfx.log](https://github.com/linrongbin16/fzfx.nvim/files/14138296/fzfx.log)
> 
> Content of the `_live_grep_last_query_cache`:
> 
> ```lua
> {"default_provider":"restricted_mode","query":"vim.fn.expand("}
> ```

hi @savchenko , 

![image](https://github.com/linrongbin16/fzfx.nvim/assets/6496887/9a10fbf9-70ee-4aed-857d-d4fbf03765f6)

when you type `vim.fn.expand(`, the rg command actually executed in the underground is

**Comment by linrongbin16** [comment]
#591  is merged into main branch, close this ticket for now. I will enable buffer previewer for more and more commands in follow PRs.
